pageserver/src/bin/pageserver.rs

```diff
@@ -271,43 +271,40 @@ fn start_pageserver(
     WALRECEIVER_RUNTIME.block_on(pageserver::broker_client::init_broker_client(conf))?;
 
     // Initialize authentication for incoming connections
-    let auth = match &conf.auth_type {
-        AuthType::Trust => None,
-        AuthType::NeonJWT => {
-            // unwrap is ok because check is performed when creating config, so path is set and file exists
-            let key_path = conf.auth_validation_public_key_path.as_ref().unwrap();
-            Some(JwtAuth::from_key_path(key_path)?.into())
-        }
-    };
-    info!("Using auth: {:#?}", conf.auth_type);
+    let mgmt_auth;
+    let compute_auth;
+    if conf.mgmt_auth_type == AuthType::NeonJWT || conf.compute_auth_type == AuthType::NeonJWT {
```

Comment:
What do you think about moving this to separate function? `start_pageserver` is already 200+ lines long. Maybe something like `auth_from_conf`

Reply:
Up to you then, I'm ok with it. Lets create an issue with tech_debt label so we wont forget about that

Reply:
opened issue #3816